gpt_index/indices/query/graph_query_engine.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any, Dict, List, Optional, Tuple
+from gpt_index.data_structs.node_v2 import IndexNode, Node, NodeWithScore
+from gpt_index.indices.postprocessor.node import BaseNodePostprocessor
+from gpt_index.indices.query.base import BaseQueryEngine
+from gpt_index.indices.query.schema import QueryBundle
+from gpt_index.indices.response.response_synthesis import ResponseSynthesizer
+from gpt_index.response.schema import RESPONSE_TYPE
+
+
+class ComposableGraphQueryEngine(BaseQueryEngine):
+    def __init__(
+        self,
+        graph: Any,
+        response_synthesizer: Optional[ResponseSynthesizer] = None,
+        retriever_id_kwargs: Optional[Dict[str, dict]] = None,
+        retriever_type_kwargs: Optional[Dict[str, dict]] = None,
+        node_postprocessors: Optional[List[BaseNodePostprocessor]] = None,
+        recursive: bool = True,
+    ) -> None:
+        from gpt_index.indices.composability.graph import ComposableGraph
+
+        assert isinstance(graph, ComposableGraph)
+        self._graph = graph
+        self._response_synthesizer = (
+            response_synthesizer or ResponseSynthesizer.from_args()
+        )
+        self._retriever_id_kwargs = retriever_id_kwargs or {}
+        self._retriever_type_kwargs = retriever_type_kwargs or {}
+        self._node_postprocessors = node_postprocessors
+
+        # additional configs
+        self._recursive = recursive
+
+    async def _aquery(self, query_bundle: QueryBundle) -> RESPONSE_TYPE:
```

Comment:
should this class implement retrieval too?

Reply:
no, I want to keep the query engine interface clean. If user wants retrieval, should just use the retriever interface.